### ISSUE TITLE
As a user, I want to click by "reaching in" with my head

### ISSUE THREAD

**Issue by labofoz** [labofoz/SeeClarke.js#1]
**The process**:
- [ ] Calculate Depth
- [ ] Fire a native click event whenever the current frames depth is lower than the one a few frames ago by X (we'll call X "click sensitivity" and make it configurable)

**Assumptions**:
- Most heads are basically similar
- The eyes and nose are always visible
- Users are not walking and clicking at the same time

**Psuedocode**:
- Find the area of the triangle between the eyes and nose. The larger the area the closer it is assumed the user is:
![image](https://user-images.githubusercontent.com/41490928/43041548-c651c8c6-8d18-11e8-8004-3bb08e06d482.png)
- Pass this raw area value as the "z-depth"

**Notes**:
- I keep saying z-depth because I don't know what I'm doing. This isn't a vector or the "z in (x, y, z) coordinates" or anything, but simply some random feature of PoseNet I believe can be correlated with the z
- It should be easy for someone with some math background to estimate the (x, y, z) in actual inches/centimeters from the webcam

**Taking it further**:
- If a person has LiDAR, then using these (x, y, z) values I should be able to trace a ray from my head to whatever object I'm looking at
- We might then be able to cut out a rectangle around that part of the webcam stream, feed it into a classifier, and be told what it is I'm pointing at
  - In other words, it should be possible for me to point at any object in my environment (literally point at the physical thing, not the image of it/myself on screen) and instantly be told what it is I'm pointing at - even without facing the camera

**Comment by labofoz** [comment]
After some quick research, I'll be using the [Shoelace formula](https://en.wikipedia.org/wiki/Shoelace_formula) to calculate the area of the triangle.

**Comment by labofoz** [comment]
The "Window Effect" demo is mostly complete, but we'll want to update SeeClarke.js to reflect our lessons learned: https://codepen.io/labofoz/full/qyrRZo

This will involve:
- [ ] Adding a "zoomFactor" helper. Because there is significant jitter (especially as you move further away) we can step the distance...kind of like electrons in an atom where they can only be within a specific orbit
- [ ] Make the zoomFactor configurable
